README.md

```diff
@@ -24,14 +23,69 @@ If you have downloaded the repo without using the `--recurse-submodules` argumen
 git submodule update --init --recursive
 ```
 
+## MQTT Config File
+
+The MQTT Agent library exposes build configuration macros that are required for building the library.
+However, the MQTT Agent library does not include its own configuration file, instead inheriting any macros defined in coreMQTT's `core_mqtt_config.h`. Lists of configurable macros and their default values can be found at the top of [core_mqtt_agent.h](source/include/core_mqtt_agent.h) and [core_mqtt_agent_command_functions.h](source/include/core_mqtt_agent_command_functions.h).
+
+To provide custom values for the configuration values, they must be either:
+* Defined in `core_mqtt_config.h` used by coreMQTT
+**OR**
+* Passed as compile time preprocessor macros
+
+## Building the Library
+
+The [mqttAgentFilePaths.cmake](mqttAgentFilePaths.cmake) file contains the information of all source files and the header include path from this repository. In addition to these files, the coreMQTT library is required to build the MQTT Agent library.
```

Comment:
```suggestion
The [mqttAgentFilePaths.cmake](mqttAgentFilePaths.cmake) file contains the information of all source files and the header include path from this repository. In addition to these files, the coreMQTT library is required to build the MQTT Agent library whose files can be obtained from [mqttFilePaths.cmake](https://github.com/FreeRTOS/coreMQTT/blob/v1.1.1/mqttFilePaths.cmake) file in the [coreMQTT](https://github.com/FreeRTOS/coreMQTT) repository.
```

Reply:
Should we really point to a particular version like this ? [mqttFilePaths.cmake](https://github.com/FreeRTOS/coreMQTT/blob/v1.1.). Since the Agent already submodules coreMQTT, is it necessary ?

Reply:
I think we could just refer to `mqttFilePaths.cmake` by name and link `coreMQTT` to the directory containing the submodule